fluent-dom/src/localization.js

```diff
@@ -55,17 +37,55 @@ export default class Localization {
    */
   async formatWithFallback(keys, method) {
     const translations = [];
+    let hasMissing = true;
+    let debugInfo = null;
+
     for (let ctx of this.ctxs) {
       // This can operate on synchronous and asynchronous
       // contexts coming from the iterator.
       if (typeof ctx.then === "function") {
         ctx = await ctx;
       }
-      const errors = keysFromContext(method, ctx, keys, translations);
-      if (!errors) {
+      const missingIds = keysFromContext(method, ctx, keys, translations);
+      if (missingIds === null) {
+        hasMissing = false;
         break;
+      } else {
+        if (debugInfo === null) {
+          debugInfo = new Map();
+        }
+        debugInfo.set(
+          ctx.locales[0],
+          missingIds
+        );
+      }
+    }
+
+    if (hasMissing) {
+      keys.forEach((key, i) => {
+        if (translations[i] === undefined) {
+          translations[i] = { value: key[0], attrs: null };
```

Comment:
This is dangerous without coordination with `overlayElement` or `applyTranslations`. Some DOM elements might not expect a value. Others might have their own children and might be only interested in translating the attributes.

Reply:
Can you please add a test for this so that we don't regress in the future?

Reply:
I have second thought on whether I should fix it here. I've seen anecdotally cases where a missing string would turn:

```
<window data-l10n-id="key1">
   ... you know, whole browser.xul
</window>
```

into:

```
<window data-l10n-id="key1">
  key1
</window>
```

So I'm going to remove it here and let's continue conversation about how to approach it.

Reply:
Yes, that's what I meant. Let's not touch it here. A test would be nice to have to make sure we don't regress in the future (like one of the iterations of this PR would have).